### DISCUSSION TITLE
Support Plex Webhooks

### DISCUSSION THREAD

**Discussion by arsaboo** [music-assistant/support#2623] [Feature requests and ideas]
When playing music using Plex, I see the play history get updated. However, Music Assistant creates no webhook that can trigger other services. For example, I am using webhooks in Plex to scrobble to Navidrome:

![image](https://github.com/user-attachments/assets/3505d0c8-b5b6-4932-945a-1004418f938c)

This works perfectly when playing music directly from Plex but does not work when using Music Assistant, which does not trigger any events in Plex.

**Comment by anatosun** [comment]
Building on this discussion, it would be valuable if Music Assistant exposed generic webhook endpoints that could accept incoming webhooks from various providers. This would enable truly real-time synchronization.

**Use Case:**
Currently, Music Assistant periodically polls providers like Plex for updates. With webhook support, these updates could happen in real-time:

- When a playlist is modified in Plex (songs added/removed/reordered)
- When new content is added to Plex libraries
- When metadata is updated in Plex

...the webhook would be triggered immediately, and Music Assistant could reflect these changes instantly rather than waiting for the next polling cycle.

**Comment by cociweb** [comment]
(Webhook requires payed license.)

**Reply by anatosun** [reply]
Yes, it would be an additional (and optional) feature for those who 'own' a Plex pass.
